cylc/flow/workflow_files.py

```diff
@@ -788,45 +790,50 @@ def remote_clean(
             "Cannot clean on remote platforms as the workflow database is "
             f"out of date/inconsistent with the global config - {exc}")
```

Comment:
Can I ask @wxtim to cast an eye over this method, regarding the retries if the clean fails on an install target using a particular platform? I think this is slightly different to intelligent host selection because we're selecting platforms to retry on instead of selecting hosts? However I also think that ultimately there will have to be a host selection per platform in the next method, `_remote_clean_cmd()`.